src/transactions/InvokeHostFunctionOpFrame.cpp

```diff
@@ -420,25 +420,24 @@ InvokeHostFunctionOpFrame::doApply(Application& app, AbstractLedgerTxn& ltx,
                     TTLEntry = ttlLtxe.current().data.ttl();
                 }
 
-                if (shouldAddEntry)
+                auto leBuf = toCxxBuf(le);
```

Comment:
I don't think we should expose (and observe) the expired temp entries. I think we need to change the flow here: read the ttl entry first, then only proceed to reading the full entry in case if it's live.

Reply:
Good idea

Reply:
Hmm should we treat the absence of a ttlEntry == absence of Soroban entry without doing the explicit check? If this doesn't hold due to (a very bad) bug, this assumption would make things much worse.

Reply:
I think that's fine. If we have missing TTL entries, we're in a very bad state regardless. If we add an invariant check here, it's both computationally expensive because we have to do more reads and expensive to devs who have to pay fees for loading unusable, expired entries. If we just load the TTL entry and stop if the entry is expired, we only have to charge devs a small fee for the TTL read instead of charging for the whole entry.

Reply:
I'm not sure if this precaution is worth the potential performance issues with this code path (or bad usability due to counting expired bytes), given that ledger state will be corrupted either way. We could also maybe add additional invariant checks at the write time, so the end result would be the same.

Reply:
> we only have to charge devs a small fee for the TTL read instead of charging for the whole entry.

FWIW we charge a flat fee for *every* entry in the footprint, whether it exists or not. TTL read or not is rather an implementation detail.

Reply:
Updated for all ops. This PR now checks the TTLEntry before deciding if it should load the actual entry for all Soroban ops.